### ISSUE TITLE
can not compile iOS app with 0.69.1

### ISSUE THREAD

**Issue by jayantnd** [facebook/react-native#34112]
### Description

/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<T> find(std::string const &key) const {

### Version

0.69.1

### Output of `npx react-native info`

System:
    OS: macOS 12.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 957.30 MB / 16.00 GB
    Shell: 3.2.57 - /bin/sh
  Binaries:
    Node: 18.4.0 - /var/folders/g6/grshvnq979b3_nl7r8j8lpmm0000gp/T/yarn--1656606680970-0.9894113721837112/node
    Yarn: 1.22.19 - /var/folders/g6/grshvnq979b3_nl7r8j8lpmm0000gp/T/yarn--1656606680970-0.9894113721837112/yarn
    npm: 8.12.1 - /usr/local/bin/npm
    Watchman: 2022.06.27.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7784292
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 15.0.2 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0 
    react-native: 0.69.1 => 0.69.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Fails after adding reactNavigation dependencies.

package.json
{
  "name": "TestApp",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint . --ext .js,.jsx,.ts,.tsx"
  },
  "dependencies": {
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.6.2",
    "react": "18.0.0",
    "react-native": "0.69.1",
    "react-native-safe-area-context": "^4.3.1",
    "react-native-screens": "^3.13.1"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "@types/jest": "^26.0.23",
    "@types/react-native": "^0.69.0",
    "@types/react-test-renderer": "^18.0.0",
    "@typescript-eslint/eslint-plugin": "^5.29.0",
    "@typescript-eslint/parser": "^5.29.0",
    "babel-jest": "^26.6.3",
    "eslint": "^7.32.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "^0.70.3",
    "react-test-renderer": "18.0.0",
    "typescript": "^4.4.4"
  },
  "resolutions": {
    "@types/react": "^18"
  },
  "jest": {
    "preset": "react-native",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ]
  }
}

### Snack, code example, screenshot, or link to a repository

[ Running command yarn react-native run-ios ]
yarn run v1.22.19
$ /Users/jayant/Src/TestApp/node_modules/.bin/react-native run-ios
info Found Xcode workspace "TestApp.xcworkspace"
info Building (using "xcodebuild -workspace TestApp.xcworkspace -configuration Debug -scheme TestApp -destination id=8BD5E894-0DCF-4848-8632-D907A8477D96")
Ignoring digest-crc-0.6.3 because its extensions are not built. Try: gem pristine digest-crc --version 0.6.3
▸ Compiling EventEmitters.cpp
▸ Compiling RNSScreenState.cpp

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<T> find(std::string const &key) const {
  ^~~~~~~~~~~~~

▸ Compiling RNSScreenStackHeaderSubviewComponentView.mm

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

In file included from /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/core/PropsParserContext.h:11:
      ^

⚠️  /Users/jayant/Library/Developer/Xcode/DerivedData/TestApp-fnqaveehuqbxsjgkvblyzkpmeifd/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNScreens.build/Objects-normal/x86_64/RNSScreenStackHeaderSubviewComponentView.dia:1:1: Could not read serialized diagnostics file: error("Failed to open diagnostics file") (in target 'RNScreens' from project 'Pods')

class Optional {
      ^

▸ Compiling RNSScreenStackHeaderConfigComponentView.mm

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<T> find(std::string const &key) const {
  ^~~~~~~~~~~~~

⚠️  /Users/jayant/Library/Developer/Xcode/DerivedData/TestApp-fnqaveehuqbxsjgkvblyzkpmeifd/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNScreens.build/Objects-normal/x86_64/RNSScreenStackHeaderConfigComponentView.dia:1:1: Could not read serialized diagnostics file: error("Failed to open diagnostics file") (in target 'RNScreens' from project 'Pods')

      ^

▸ Compiling RNSScreenStackComponentView.mm

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<T> find(std::string const &key) const {
  ^~~~~~~~~~~~~

⚠️  /Users/jayant/Library/Developer/Xcode/DerivedData/TestApp-fnqaveehuqbxsjgkvblyzkpmeifd/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNScreens.build/Objects-normal/x86_64/RNSScreenStackComponentView.dia:1:1: Could not read serialized diagnostics file: error("Failed to open diagnostics file") (in target 'RNScreens' from project 'Pods')

      ^

▸ Compiling RNSScreenShadowNode.cpp

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<T> find(std::string const &key) const {
  ^~~~~~~~~~~~~

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/AccessibilityPrimitives.h:54:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<std::string> label{};
      ^

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/AccessibilityPrimitives.h:108:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<int> min;
      ^

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/AccessibilityPrimitives.h:109:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<int> max;
      ^

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/AccessibilityPrimitives.h:110:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<int> now;
      ^

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/AccessibilityPrimitives.h:111:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<std::string> text{};
      ^

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/primitives.h:78:21: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  using OptionalT = std::optional<T>;
      ^

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/primitives.h:139:21: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  using OptionalT = std::optional<T>;
      ^

⚠️  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:114:20: empty paragraph passed to '@param' command [-Wdocumentation]

   * @param progress
      ^

⚠️  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:115:15: empty paragraph passed to '@param' command [-Wdocumentation]

   * @param lhs
     ~~~~~~~~~~~~~~^



⚠️  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:116:15: empty paragraph passed to '@param' command [-Wdocumentation]

   * @param rhs
     ~~~~~~~~~^



⚠️  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:117:13: empty paragraph passed to '@return' command [-Wdocumentation]

   * @return
     ~~~~~~~~~^



⚠️  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:114:13: parameter 'progress' not found in the function declaration [-Wdocumentation]

   * @param progress
     ~~~~~~~^



❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/renderer/components/view/ViewProps.h:60:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<int> zIndex{};
            ^~~~~~~~



⚠️  /Users/jayant/Library/Developer/Xcode/DerivedData/TestApp-fnqaveehuqbxsjgkvblyzkpmeifd/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNScreens.build/Objects-normal/x86_64/RNSScreenShadowNode.dia:1:1: Could not read serialized diagnostics file: error("Failed to open diagnostics file") (in target 'RNScreens' from project 'Pods')


      ^


▸ Compiling RNSScreenController.mm

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<T> find(std::string const &key) const {
  ^~~~~~~~~~~~~



⚠️  /Users/jayant/Library/Developer/Xcode/DerivedData/TestApp-fnqaveehuqbxsjgkvblyzkpmeifd/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNScreens.build/Objects-normal/x86_64/RNSScreenController.dia:1:1: Could not read serialized diagnostics file: error("Failed to open diagnostics file") (in target 'RNScreens' from project 'Pods')


      ^


▸ Compiling RNSScreenComponentView.mm

❌  /Users/jayant/Src/TestApp/ios/Pods/Headers/Public/React-Fabric/react/utils/ContextContainer.h:95:3: no template named 'optional' in namespace 'std'; did you mean 'folly::Optional'?

  std::optional<T> find(std::string const &key) const {
  ^~~~~~~~~~~~~



⚠️  /Users/jayant/Library/Developer/Xcode/DerivedData/TestApp-fnqaveehuqbxsjgkvblyzkpmeifd/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNScreens.build/Objects-normal/x86_64/RNSScreenComponentView.dia:1:1: Could not read serialized diagnostics file: error("Failed to open diagnostics file") (in target 'RNScreens' from project 'Pods')


      ^


error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening TestApp.xcworkspace.
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
sh-3.2$

**Comment by jayantnd** [comment]
issue resolved after upgrading the dependency packages.
Closing this issue.

**Comment by leonchabbey** [comment]
@jayantnd What did you actually do to fix it? I'm trying to run [rn-tester (0.68.x)](https://github.com/facebook/react-native/tree/0.68-stable/packages/rn-tester) but once I try to build it, I get:

![Screenshot 2023-01-03 at 19 49 30](https://user-images.githubusercontent.com/12344434/210422101-dd45eaa1-5386-432e-b069-ed718256e975.png)

OS: macOS 13.1
Xcode: 14.2/14C18

**Comment by Mourad-Tlili** [comment]
Any updates ?
